crates/pallets/starknet/src/tests/invoke_tx.rs

```diff
@@ -525,3 +529,127 @@ fn test_verify_nonce_in_unsigned_tx() {
+            Felt252Wrapper::from_hex_be("0x1c02b663e928ed213d3a0fa206efb59182fa2ba41f5c204daa56c4a434b53e5").unwrap();
+
+        let mut transaction = DeclareTransactionV2 {
+            sender_address: account_addr.into(),
+            class_hash: transaction_revert_class_hash,
+            compiled_class_hash: transaction_revert_compiled_class_hash,
+            nonce: Felt252Wrapper::ZERO,
+            max_fee: u128::MAX,
+            signature: vec![],
+            offset_version: false,
+        };
+
+        let chain_id = Starknet::chain_id();
+        let transaction_hash = transaction.compute_hash::<<MockRuntime as Config>::SystemHash>(chain_id, false);
+        transaction.signature = sign_message_hash(transaction_hash);
+
+        // validate declare transaction
+        assert_ok!(Starknet::validate_unsigned(
+            TransactionSource::InBlock,
+            &crate::Call::declare {
+                transaction: transaction.clone().into(),
+                contract_class: transaction_revert_class.clone()
+            },
+        ));
+
+        assert_ok!(Starknet::declare(none_origin, transaction.into(), transaction_revert_class.clone()));
+        assert_eq!(
+            Starknet::contract_class_by_class_hash(ClassHash::from(transaction_revert_class_hash)).unwrap(),
+            transaction_revert_class
+        );
+
+        let salt = Felt252Wrapper::ZERO;
+
+        let deploy_transaction = InvokeTransactionV1 {
+            sender_address: account_addr.into(),
+            signature: vec![],
+            nonce: Felt252Wrapper::ONE,
+            calldata: vec![
+                Felt252Wrapper::ONE,
+                Felt252Wrapper::from_hex_be(UDC_ADDRESS).unwrap(), // udc address
+                Felt252Wrapper::from_hex_be(UDC_SELECTOR).unwrap(), // deployContract selector
+                Felt252Wrapper::from_hex_be("0x4").unwrap(),       // calldata len
+                transaction_revert_class_hash,                     // contract class hash
+                salt,                                              // salt
+                Felt252Wrapper::ONE,                               // unique
+                Felt252Wrapper::ZERO,                              // constructor calldata len
+            ],
+            max_fee: u128::MAX,
+            offset_version: false,
+        };
+
+        // validate invoke transaction
+        assert_ok!(Starknet::validate_unsigned(
+            TransactionSource::InBlock,
+            &crate::Call::invoke { transaction: deploy_transaction.clone().into() },
+        ));
+
+        let contract_address: FieldElement = get_udc_deployed_contract_address(
+            salt.into(),
+            transaction_revert_class_hash.into(),
+            &UdcUniqueness::Unique(UdcUniqueSettings {
+                deployer_address: FieldElement::from_hex_be(
+                    "0x0642a8b9e2c6cc3a9ddb84575123f262a21415f78db453b0625d889e1e06ac32",
```

Comment:
```suggestion
                    account_addr,
```

Reply:
I had tried using `account_adr` instead of hardcoding the value. But since `account_adr` is of type `ContractAddress` and `deployer_address` expects value of type `FieldElement`, I am not sure how to convert ContractAddress to FieldElement. Need help here @EvolveArt

Reply:
Can you try account_adr.into(()

Reply:
account_addr.into() doesn't work, it says the following error:

<img width="643" alt="Screenshot 2024-04-04 at 12 21 07 AM" src="https://github.com/keep-starknet-strange/madara/assets/26214362/7752f4df-886e-49b0-9d0b-06b69d3afe7b">

Reply:
My bad, can you try account_adr.0.into()

Reply:
Thanks, this `account_addr.0.0.into()` worked